PostProcessing/master.py

```diff
@@ -91,22 +102,71 @@ def HL_fitting_function(infile, outfilename, func_name="MultiGauss",
         workers.close()
 
         # Unravel results into output grids
-        params, obs_err, chi_grid = Posproc.results_to_grid(results, len(lats), len(lons))
+        params, obs_err, rss_func_grid, rss_mean_grid, dof = (
+                        Posproc.results_to_grid(results, len(lats), len(lons)))
 
         # Plot some results if requested
         if plot != None:
            print("MESSAGE: Plotting results - data versus fitting: " + str(depth[lev]) + " m")
            Plots.plot_data_vs_fitting(outfig, plot, x_val, cors, var, obs_err, lats, lons,
                                        depth[lev], params, func_name, num_funcs, lenscale)
-                                      
+
+        # If requested perform F-test comparing to mean
+        if f_test:
+            if (func_name == "MultiGauss"):
+                num_params = 2 * num_funcs
+            elif (func_name == "MultiGauss_Fixed"):
+                num_params = num_funcs
+            else:
+                raise ValueError("Cannot calculate num_params for "
+                                 + f"function={func_name}")
+            p_val = f_test_pvalue(rss_func_grid, rss_mean_grid, num_params,
+                           num_params + dof)
+        else:
+            p_val = None
+
         print(f"MESSAGE: Writing data to netcdf file: {outfile}")
         if lev == 0:
            for param in range(0, len(params)):
                # Define netcdf variables from fitting function results
                IO.nc_define_vars(outfile, arg_lists[0]["func"].param_names()[param],
                                  'f', ("depth","latitude","longitude"))
-      
+
         # Add variables to netcdf
-        IO.ncwrite_output(outfile, arg_lists[0]["func"], chi_grid, obs_err, params, lev) 
+        IO.ncwrite_output(outfile, arg_lists[0]["func"], rss_func_grid,
+                          rss_mean_grid, dof, obs_err, params, lev, p_val=p_val)
     
     outfile.close()
+
+def f_test_pvalue(rss_func, rss_mean, num_param, num_points):
```

Comment:
Should we move this function "f_test_pvalue" to posproc.py or any other python module? I feel like the master.py should contain only the main code structure and all other functions should be located in different modules.

Reply:
I've created a new file StatisticTests.py and moved the test into there.